Makefile

```diff
@@ -99,6 +99,10 @@ endif
 # use a separate waku discv5 network with `protocol-id="d5waku"`
 NIM_PARAMS := $(NIM_PARAMS) -d:discv5_protocol_id:d5waku
 
+# git version for JSON RPC call
+GIT_VERSION := "$(shell git describe --abbrev=6 --always --tags)"
+NIM_PARAMS := $(NIM_PARAMS) -d:git_version:\"$(GIT_VERSION)\"
```

Comment:
I would not recommend setting variables that call commands with `shell` using immediate assignment(`:=`), you should be using lazy assignment(`=`) instead to avoid calling this command EVERY SINGLE TIME `make` is called.

For more details see:

* https://www.gnu.org/software/make/manual/html_node/Flavors.html
* https://stackoverflow.com/questions/448910/

Reply:
I could do that for `GIT_VERSION`, but `NIM_PARAMS` is already used quite a lot with `:=` (and doing a single assignment with = for a recursive variable does not work). So `GIT_VERSION` would still be eagerly evaluated.

`NIM_PARAMS += -d:git_version:\"$(GIT_VERSION)\"` I did this in another PR before, but this will evaluate in the same way the other uses of `NIM_PARAMS` do and we reverted that to be consistent with the other assignments.

Should we rethink this for the whole Makefile and see which vars can actually be lazily evaluated? But imo, this should be in a new PR. Wdyt?

Reply:
I will put this on my list :).